### PR TITLE
CLOUDP-280222: Private Endpoint custom resource API

### PR DESCRIPTION
**Pull request by helderjs**
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes-template.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: PROJECT [hunk 1]
```diff
@@ -95,4 +95,13 @@ resources:
   kind: AtlasSearchIndexConfigs
   path: github.com/mongodb/mongodb-atlas-kubernetes/api/v1
   version: v1
+- api:
```
Comment (s-urbaniak) [inline PROJECT:98]: remind me please: what's this file for?

Comment (helderjs) [reply PROJECT:98]: it's Kubebuilder configuration

File: pkg/api/v1/atlascustomresource.go [hunk 2]
```diff
@@ -29,6 +29,7 @@ var _ AtlasCustomResource = &AtlasStreamInstance{}
 var _ AtlasCustomResource = &AtlasStreamConnection{}
 var _ AtlasCustomResource = &AtlasSearchIndexConfig{}
 var _ AtlasCustomResource = &AtlasBackupCompliancePolicy{}
+var _ AtlasCustomResource = &AtlasPrivateEndpoint{}
```
Comment (josvazg) [inline pkg/api/v1/atlascustomresource.go:33]: Is this here only to force an error if any of those types is not an AtlasCustomResource?

File: pkg/api/v1/atlasprivateendpoint_types.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+	for _, o := range options {
+		v := o.(AtlasPrivateEndpointStatusOption)
+		v(&pe.Status)
+	}
+}
+
+func init() {
+	SchemeBuilder.Register(&AtlasPrivateEndpoint{}, &AtlasPrivateEndpointList{})
+}
```
Comment (josvazg) [inline pkg/api/v1/atlasprivateendpoint_types.go:167]: Seems this is missing a credentials implementation:

```go
func (p AtlasPrivateEndpoint) Credentials() *api.LocalObjectReference {
	return p.Spec.Credentials()
}
```

File: pkg/api/v1/atlascustomresource.go (hunk 2 above)
Comment (helderjs) [reply pkg/api/v1/atlascustomresource.go:33]: Exactly

File: pkg/api/v1/atlasprivateendpoint_types.go [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+	pe.Status.Conditions = conditions
+	pe.Status.ObservedGeneration = pe.ObjectMeta.Generation
+
+	for _, o := range options {
+		v := o.(status.AtlasPrivateEndpointStatusOption)
+		v(&pe.Status)
+	}
+}
+
+func init() {
```
Comment (s-urbaniak) [inline pkg/api/v1/atlasprivateendpoint_types.go:155]: nit: please put `init()` functions on top of the file

File: pkg/api/v1/status/atlasprivateendpoint.go [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	ConnectionName string `json:"connectionName,omitempty"`
+	// GCPForwardingRules is the status of the customer GCP private endpoint(forwarding rules)
+	GCPForwardingRules []GCPForwardingRule `json:"gcpForwardingRules,omitempty"`
+	// InterfaceStatus is the state of the private endpoint interface
+	Status string `json:"InterfaceStatus,omitempty"`
+	// Error is the description of the failure occurred when configuring the private endpoint
+	Error string `json:"error,omitempty"`
+}
+
+type GCPForwardingRule struct {
```
Comment (s-urbaniak) [inline pkg/api/v1/status/atlasprivateendpoint.go:55]: this is missing godoc

**Review by s-urbaniak — APPROVED** [review]
really just nits, else lgtm

File: pkg/api/v1/status/atlasprivateendpoint.go (hunk 5 above)
Comment (helderjs) [reply pkg/api/v1/status/atlasprivateendpoint.go:57]: done

File: pkg/api/v1/atlasprivateendpoint_types.go (hunk 4 above)
Comment (helderjs) [reply pkg/api/v1/atlasprivateendpoint_types.go]: done

**Pull request merged**